### PR TITLE
ibmcloud: Add variable for primary_nic state and nm_dns management

### PR DESCRIPTION
**Pull request by nhicher**
With this change, user can specify on local-overrides.yaml:

- primary_nic_state to change the required state for the
  interface (eg: primary_nic_state: up)
- set the value for NetworkManager dns config (eg: dns: none)

### CODE REVIEW

File: playbooks/prepare_host.yaml [hunk 1]
```diff
@@ -121,15 +120,14 @@
   # resolv.conf
   - name: Enable NetworkManager ownership of resolv.conf
     when:
-      - false
       - ansible_facts.distribution == 'CentOS'
     block:
       - name: Set main/dns=default in NetworkManager
```
Comment (mandre) [inline playbooks/prepare_host.yaml:125]: Perhaps we should update the task name too?
```suggestion
      - name: Set main/dns in NetworkManager
```

**Review by mandre — COMMENTED** [review]
Hi @nhicher , it would probably be easier if you create a new PR against master. It took me some time to see you created a PR against an open PR. I didn't even know you could do this :D

File: playbooks/prepare_host.yaml (hunk 1 above)
Comment (nhicher) [reply playbooks/prepare_host.yaml:125]: Hello @mandre,
I test dev-install with the branch @EmilienM created for ibm baremetal instance. I have probably some others stuffs to add.
Thanks for the review

**Comment by EmilienM** [comment]
I'm not sure why we needed this just on IBM cloud, I'll close this now. Re-open if needed.

**Pull request closed without merging**